client/python/lib/ovmsclient/tfs_compat/grpc/serving_client.py

```diff
@@ -105,7 +116,111 @@ def get_model_status(self, request):
+            private_key=client_key, certificate_chain=client_cert)
+
+            channel = grpc.secure_channel(grpc_address, creds)
+        else:
+            channel = grpc.insecure_channel(grpc_address)
+
+        prediction_service_stub = prediction_service_pb2_grpc.PredictionServiceStub(channel)
+        model_service_stub = model_service_pb2_grpc.ModelServiceStub(channel)
+        stubs = {
+            "prediction_service_stub" : prediction_service_stub,
+            "model_service_stub" : model_service_stub
+        }
+
+        return cls(channel, stubs)
+
+def _prepare_certs(client_key, client_cert, server_cert):
+    
+    if client_key is not None:
+        with open(client_key, 'rb') as key:
+            client_key = key.read()
+        try:
+            p_key = OpenSSL.crypto.load_privatekey(OpenSSL.crypto.FILETYPE_PEM, client_key)
+        except OpenSSL.crypto.Error as e_info:
+            raise ValueError('client key file is not valid')
+
+    if client_cert is not None:
+        with open(client_cert, 'rb') as ca:
+            client_cert = ca.read()
+        try:
+            c_cert = OpenSSL.crypto.load_certificate(
+                OpenSSL.crypto.FILETYPE_PEM,
+                client_cert
+            )
+        except OpenSSL.crypto.Error as e_info:
+            raise ValueError('client certificate file is not valid')
+        
+    with open(server_cert, 'rb') as f:
+        server_cert = f.read()
+    try:
+        s_cert = OpenSSL.crypto.load_certificate(
+            OpenSSL.crypto.FILETYPE_PEM,
+            server_cert
+        )
+    except OpenSSL.crypto.Error as e_info:
+        raise ValueError('server certificate file is not valid')
+
+    return client_key, client_cert, server_cert
+
+def _check_config(config):
+    
+    if 'address' not in config or 'port' not in config:
+        raise Exception('The minimal config must contain address and port')
+
+    _check_address(config['address'])
+
+    _check_port(config['port'])
+
+    if 'tls_config' in config:
+        _check_tls_config(config['tls_config'])
+    
+def _check_address(address):
+
+    if not isinstance(address, str):
+        raise TypeError(f'address type should be string, but is {type(address).__name__}')
+
+    if address != "localhost" and not ipv4(address) and not domain(address):
+        raise ValueError(f'address is not valid')
+    
+
+def _check_port(port):
+    
+    if not isinstance(port, int):
+        raise TypeError(f'port type should be int, but is type {type(port).__name__}')
+
+    if port.bit_length() > 16 or port < 0:
+        raise ValueError(f'port should be in range <0, {2**16-1}>')
+
+def _check_tls_config(tls_config):
+
+    if 'server_cert_path' not in tls_config:
+        raise ValueError(f'server_cert_path is not defined in tls_config')
+    if 'client_key_path' in tls_config and 'client_cert_path' not in tls_config or 'client_cert_path' in tls_config and 'client_key_path' not in tls_config:
```

Comment:
It's a quite long condition. I wonder if something like that would be more readable:
```
if 'client_key_path' in tls_config != 'client_cert_path' in tls_config:
    raise(...)
```
Up to you 😄